### PR TITLE
Display error message on failed logins attempts

### PR DESCRIPTION
**Pull request by thomasdseao**

### CODE REVIEW

File: src/app/modules/landing/modules/auth/components/login/login.component.ts [hunk 1]
```diff
@@ -40,7 +41,13 @@ export class LoginComponent {
...
           this.store.dispatch(new Navigate([returnUrl || '/']));
         },
-        (err: HttpErrorResponse) => this.store.dispatch(new ToasterError(extractError(err))),
+        (err: HttpErrorResponse) => {
+          this.store.dispatch(new ToasterError(extractError(err)));
+          event.target.disabled = false;
+        },
+        () => {
+          event.target.disabled = false;
+        }
```
Comment (ozergul) [inline src/app/modules/landing/modules/auth/components/login/login.component.ts]: you have to subscribe like this:
```
new AuthLogin({ ...}).subscribe({
   error: err => {....}
})
```

File: src/app/modules/landing/modules/auth/components/login/login.component.html [hunk 2]
```diff
@@ -15,7 +15,7 @@
   <div class="col-lg-4 col-12">
     <div class="card">
       <div class="card-body">
-        <form [formGroup]="form" (keydown.enter)="submitForm()">
+        <form [formGroup]="form" (keydown.enter)="submitForm($event)">
```
Comment (ozergul) [inline src/app/modules/landing/modules/auth/components/login/login.component.html]: no need to send $event. you can create a flag in ts like this:
`submitClicked = false;`
When click, set it to true
In subscribe set it to false

in submitForm() method control it

if(!this.submitClicked) return;

this.submitClicked = true;

**Review by ozergul — CHANGES_REQUESTED** [review]

File: src/app/modules/landing/modules/auth/components/login/login.component.html (hunk 2 above)
Comment (thomasdseao) [reply src/app/modules/landing/modules/auth/components/login/login.component.html]: Yes good idea, done :)

File: src/app/modules/landing/modules/auth/components/login/login.component.ts [hunk 3]
```diff
@@ -43,10 +43,10 @@ export class LoginComponent {
         },
         (err: HttpErrorResponse) => {
           this.store.dispatch(new ToasterError(extractError(err)));
-          event.target.disabled = false;
+          this.submitClicked = false;
         },
```
Comment (ozergul) [inline src/app/modules/landing/modules/auth/components/login/login.component.ts]: you have to subscribe to AuthLogin action

**Review by ozergul — CHANGES_REQUESTED** [review]

File: src/app/modules/landing/modules/auth/components/login/login.component.ts (hunk 3 above)
Comment (thomasdseao) [reply src/app/modules/landing/modules/auth/components/login/login.component.ts]: Ok but i am already in the subscribe of the dispatch of AuthLogin action, what do you mean ?

File: src/app/modules/landing/modules/auth/components/login/login.component.ts [hunk 4]
```diff
@@ -40,7 +41,13 @@ export class LoginComponent {
           const { returnUrl } = this.route.snapshot.params;
           this.store.dispatch(new Navigate([returnUrl || '/']));
         },
-        (err: HttpErrorResponse) => this.store.dispatch(new ToasterError(extractError(err))),
+        (err: HttpErrorResponse) => {
```
Comment (ozergul) [inline src/app/modules/landing/modules/auth/components/login/login.component.ts:44]: new AuthLogin({ ...}).subscribe({
   error: err => {....}
})

**Review by ozergul — CHANGES_REQUESTED** [review]

File: src/app/modules/landing/modules/auth/components/login/login.component.ts (hunk 4 above)
Comment (thomasdseao) [reply src/app/modules/landing/modules/auth/components/login/login.component.ts:44]: it is not possible to subscribe on AuthLogin directly

**Pull request closed without merging**